tests/frontend/components/test_build_input_section.py

```diff
@@ -0,0 +1,178 @@
+"""Test function build_input_field_section."""
+
+import dash_bootstrap_components as dbc
+import pytest
+import ssb_dash_components as ssb  # type: ignore[import-untyped]
+from datadoc_model import model
+
+from datadoc import enums
+from datadoc.enums import SupportedLanguages
+from datadoc.frontend.components.builders import build_input_field_section
+from datadoc.frontend.fields.display_base import get_enum_options_for_language
+from datadoc.frontend.fields.display_variables import OBLIGATORY_VARIABLES_METADATA
+from datadoc.frontend.fields.display_variables import OPTIONAL_VARIABLES_METADATA
+
+NORSK_BOKMÅL = "nb"
+NORSK_NYNORSK = "nn"
+ENGLISH = "en"
+
+INPUT_FIELD_SECTION = [
+    build_input_field_section(
+        OBLIGATORY_VARIABLES_METADATA,
+        model.Variable(short_name="hoveddiagnose"),
+        NORSK_NYNORSK,
+    ),
+    build_input_field_section(
+        OBLIGATORY_VARIABLES_METADATA,
+        model.Variable(short_name="pers_id"),
+        NORSK_BOKMÅL,
+    ),
+    build_input_field_section(
+        OBLIGATORY_VARIABLES_METADATA,
+        model.Variable(short_name="ber_bruttoformue"),
+        ENGLISH,
+    ),
+    build_input_field_section(
+        OBLIGATORY_VARIABLES_METADATA,
+        model.Variable(short_name="sykepenger"),
+        NORSK_BOKMÅL,
+    ),
+]
+
+
+def test_build_input_field_section_no_input_return_empty_list():
+    input_section = build_input_field_section(
+        [],
+        "",
+        "",
+    )
+    assert input_section.children == []
+
+
+@pytest.mark.parametrize(
+    ("input_field_section"),
+    INPUT_FIELD_SECTION,
+)
+def test_build_input_fields_props_input(input_field_section):
+    variable_input_field_for_name = input_field_section.children[0]
+    assert variable_input_field_for_name.type == "text"
+    assert variable_input_field_for_name.value is None
+    input_value = "Statistics"
+    variable_input_field_for_name.value = input_value
+    assert variable_input_field_for_name.value == input_value
```

Comment:
It doesn't look like this part with setting the value is really testing anything, maybe we should remove it?